### PR TITLE
Xenobio Rework Part ????: Xenoflora

### PR DESCRIPTION
**Pull request by SELFHELL**
## About The Pull Request

This PR is an atomized and (for now partially) ported version of Smartkar's  Xenobiology rework, this PR specifically seeks to add the Xenoflora system to the game. 

This PR adds the Xenoflora Pod, a 2x2 tile structure that uses both atmospherics and plumbing to grow Xenoflora/Alien plants

Compared to a player for referenced.
Demos

Some of the plants within the pod + interface

more

TGUIshit

Got it

## Why It's Good For The Game

First and foremost, this PR adds a portion of https://hackmd.io/aHChvaHySMaKZ5LF-EJ7Nw and is one of the major aspects of a xenobiology rework.

Due to the atmospheric and chemical requirements, the far more unique plant aspects, along with its eventually practical applications in Xenobiology, Xenoflora (hopefully) makes itself distinct from our existing botany system. 

Xenoflora as a system will be heavily integrated with Xenofauna and the rest of Xenobiology in the future.

As for the cargo bit, I plan to properly map out Xenoflora pods and establish proper Xenoflora seed distribution methods in a future PR but this approach serves to give players access to the system while the rest of the rework is being completed.

## Credits

A large majority of credit for this PR goes to Smartkar, a very talented coder who left the community a while back before finishing his xenobiology rework, I just ported his work to modern code and added some content here and there. 

## Changelog

:cl: Smartkar, SELFHELL
add: Xenoflora: grow alien plants with varied and unique growth conditions. 
add: Xenoflora Starter Kit in Cargo
add: Xenoflora Sprites + Associated TGUI
/:cl:

### CODE REVIEW

**Comment by SELFHELL** [comment]
oh boy.

**Comment by Paxilmaniac** [comment]
The big growing tank thing goes hard

**Comment by Shadowflame909** [comment]
Another neat mechanic locked behind atmos wizards

File: code/modules/xenobiology/xenoflora_pod/xenoflora_pod.dm [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+/obj/machinery/atmospherics/components/binary/xenoflora_pod/proc/inject_gases()
+	if(internal_gases.return_volume() >= XENOFLORA_MAX_MOLES)
+		return
+
+	var/datum/gas_mixture/input_gases = airs[2]
+	for(var/gas_type in plant.required_gases)
+		if(!input_gases.gases[gas_type] || !input_gases.gases[gas_type][MOLES])
+			continue
+
+		var/pump_amount = min(input_gases.gases[gas_type][MOLES], XENOFLORA_MAX_MOLES - internal_gases.return_volume())
```
Comment (Pickle-Coding) [inline code/modules/xenobiology/xenoflora_pod/xenoflora_pod.dm:157]: Total moles, not volume.

**Comment by DaCoolBoss** [comment]
I don't like how this interacts with chemistry and atmospherics, but not botany. Feels really weird to have a plant-based system that doesn't involve the 'plant job' at all.

**Comment by Pickle-Coding** [comment]
I think it would be better if the gases would just sit inside the machine, and filtering has to be done manually with existing atmos tools, rather than having the machine do the pumping and shit.

**Comment by optimumtact** [comment]
It's hard to justify this in isolation without all the proposed changes to the rest of xenobiology around slimes, otherwise you're just adding new content without fixing any of the prior issues.

Also this being your first PR doesn't exactly inspire confidence you have a vision/plan for this beyond what smartkar had drafted out.

**Comment by ATH1909** [comment]
not a big fan of this being dependent on both atmos AND chemistry; cytology was dependent on just chemistry (and botany), and look how that turned out.

~~I might just be biased bc I don't want to learn atmos, though.~~

**Comment by GuillaumePrata** [comment]
It makes sense for people to get spooked about Atmos being added here, but the requirements are so mild anyone that can do the most basics of ordnance bombs will manage.

If this was Botany content (which honestly, would be cute as a joint venture between both departments) then it would be annoying as you would 100% be "forced to coop".
For science the atmos requirements are so mild that learning Xenoflora might actually be a good first step into ordnance.

File: code/modules/xenobiology/xenoflora_pod/xenoflora_plant.dm [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	progress += 1
+
+	if(progress >= max_progress && stage < max_stage)
+		progress = 0
+		stage += 1
+		parent_pod.update_icon()
+
+	if(LAZYLEN(produced_gases))
+		for(var/gas_type in produced_gases)
+			if(parent_pod.internal_gases.return_volume() >= XENOFLORA_MAX_MOLES)
```
[the remark this answers is not in the document]
Comment (SELFHELL) [reply code/modules/xenobiology/xenoflora_pod/xenoflora_plant.dm]: Good point

**Comment by Wallemations** [comment]
Oh hey I remember this
Sprite still holds up but fix the pipes so they're properly underneath

**Comment by AtomTheProphet** [comment]
Hey. I really like this! But wouldn't it make more sense for it to use microbes rather than plants? Like a bioreactor.

**Comment by AtomTheProphet** [comment]
I like it but honestly it'd make more sense to have a bioreactor that uses microorganisms rather than plants to produce chemicals or other substances.

**Comment by ArcaneMusic** [comment]
Personally weighing in, I think this makes a lot more sense as a kind of macro project to drop into hydroponics for them to work towards later in the shift, whose produce could be beneficial to xenobio at a later date, than for it to be a cornerstone for xenobio and feature a tiny bit of hydroponics.

Likely based on smartkar's work, I've liked the idea of plants that can have significant atmospherics effects, so believe me there are parts of these changes that I'd be happy to support. However, this really does feel like a vertical slice of a complete work in a way that neither feels very modular, nor very cohesive. To that respect, I'd agree with oranges here. Take the time to draft up a design document, aim to scope out what changes could be made with xenoflora specifically, without necessarily breaking too far into necessary changes for xenobio. Hell, if you tweaked some of the requirements, this could actually work quite well as a station goal based on the chemical requirements and gas requirements (Are we still doing those?).

**Comment by the-orange-cow** [comment]
This almost feels like it should be an extension of cytology, rather than a new independent system.

**Comment by Xkallubar** [comment]
This shouldn't be added on it's own.

**Comment by Loafin34** [comment]
can you be able to grow the potted plants? like the blue one with the black hexagons and the yellow one

**Comment by SELFHELL** [comment]
Alright, seeing all the feedback I have been given + a chat with the orange man, ill be closing this PR and eventually (hopefully) coming back with a much larger PR to create a much more complete package with some of the rework's other systems connected to xenoflora.

**Pull request closed without merging**